Doctrine/PHPCR/ElasticaToDocumentTransformer.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace FOS\ElasticaBundle\Doctrine\PHPCR;
+
+use FOS\ElasticaBundle\Doctrine\AbstractElasticaToModelTransformer;
+
+/**
+ * Maps Elastica documents with Doctrine objects
+ * This mapper assumes an exact match between
+ * elastica documents ids and doctrine object ids
+ */
+class ElasticaToDocumentTransformer extends AbstractElasticaToModelTransformer
+{
+    /**
+     * Fetch objects for theses identifier values
+     *
+     * @param array $identifierValues ids values
+     * @param Boolean $hydrate whether or not to hydrate the objects, false returns arrays
+     * @return array of objects or arrays
+     */
+    protected function findByIdentifiers(array $identifierValues, $hydrate)
+    {
+        // Special case where model is interface. But maybe remove because negative performance impact
+        $reflectionClass = new \ReflectionClass($this->objectClass);
+        if ($reflectionClass->isInterface()) {
+            return iterator_to_array(
+                $this->registry
+                    ->getManager()
+                    ->findMany(null, $identifierValues)
+            );
+        }
+
+        // General case
+        return iterator_to_array(
+            $this->registry
+                ->getManager()
+                ->getRepository($this->objectClass)
```

Comment:
in terms of phpcr-odm, this means that objects of this class or extending classes will be found. which sounds like a good idea to me. (so just to make sure you are aware of that)